brigadier/projects/views.py

```diff
@@ -1,17 +1,38 @@
 from django.views import generic
 from django.urls import reverse_lazy
+from django.db.models import Count, Q, FloatField
+from django.db.models.functions import Cast
 
-from .models import Project, Task
+from .models import Project, Task, COMPLETED
 from .forms import ProjectModelForm, TaskModelForm
 
 
+def annotate_completed_percentage(field_list):
+    """todo()
+
+    """
+    return Project.objects.values(*field_list).annotate(
+            tasks_count=Count('task'),
+            percentage_completed=Cast(Count(
+                'task',
+                filter=Q(task__status=COMPLETED),
+            ), FloatField()) / Cast(Count('task'), FloatField()) * 100
+        )
+
+
 class ProjectListView(generic.ListView):
     """View displays the list of projects.
 
     """
-    model = Project
     template_name = 'project_list.html'
-    ordering = 'deadline'
+
+    def get_queryset(self):
+        """todo()
+
+        """
+        return annotate_completed_percentage(
+            ['id', 'project_name', 'deadline', 'budget', 'closed']
+        ).order_by('deadline')
```

Comment:
unnecessary refactoring

Reply:
Why? It was changed to avoid duplicate code in detail and list views.

Reply:
then it should be written completely different
don't try to shoot in your foot